xarray/tests/test_backends.py

```diff
@@ -992,6 +999,46 @@ def test_encoding_unlimited_dims(self):
             with pytest.warns(UserWarning):
                 ds.to_netcdf(tmp_file, engine='h5netcdf', unlimited_dims=['y'])
 
+class OpenMFDatasetTest(TestCase):
+    def validate_open_mfdataset_large_num_files(self, engine, nfiles=2000):
```

Comment:
I should note there is a third issue: I have a stack of context managers wrapped inside a context manager if I go this approach, which is confusing too so we will get clearer code by making slow/fast tests.  But, this is obviously at least partially unsettling because of the runtime cost.